app/models/ont/run.rb

```diff
@@ -10,57 +10,61 @@ class Run < ApplicationRecord
     NAME_PREFIX = 'ONTRUN-' # Used for generating a unique experiment name for the run.
 
     # This association creates a link to the instrument this run belongs to.
-    # We are setting to avoid automatic detection of inverse association from the instrument.
-    # XXX: Do we need a default instrument to initialise the run with?
-    belongs_to :instrument,
-               class_name: 'Ont::Instrument',
-               foreign_key: :ont_instrument_id,
-               inverse_of: false
+    # We are setting inverse_of to false to avoid detection of inverse association.
+    belongs_to :instrument, class_name: 'Ont::Instrument', foreign_key: :ont_instrument_id,
+                            inverse_of: false
 
     has_many :flowcells, foreign_key: :ont_run_id, inverse_of: :run, dependent: :destroy
 
     accepts_nested_attributes_for :flowcells, allow_destroy: true
 
     scope :active, -> { where(deactivated_at: nil) }
 
-    validate :check_max_number_of_flowcells, :check_flowcell_position, :check_flowcell_pool
-
-    def check_flowcell_position
-      return if instrument.blank?
-
-      position_set = Set.new
-      flowcells.each do |flowcell|
-        next if flowcell.position.blank?
-
-        if flowcell.position < 1 || flowcell.position > instrument.max_number_of_flowcells
-          errors.add(:flowcells, "position #{flowcell.position} is out of range for the instrument")
-        elsif position_set.include? flowcell.position
-          errors.add(:flowcells, "position #{flowcell.position} is duplicated in the same run")
-        else
-          position_set.add(flowcell.position)
-        end
+    # run flowcells validations
+    validates :flowcells, presence: true
+    validates :flowcells, length: {
```

Comment:
Again I am not entirely sure why you need a block for this?. The message is fixed so can't you just have the message as a string?

Reply:
I have removed the lambda block for messages and added rubocop:disable Rails/I18nLocaleTexts .